### DISCUSSION TITLE
cannot import name 'ParamSpec' from 'typing_extensions' -

### DISCUSSION THREAD

**Discussion by programmeddeath1** [fastapi/fastapi#8481] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import routing
```

### Description

On importing routing from Fast API in the starlette.datastructures concurrency package, -> from typing_extension import ParamSpec is giving the following error.
`  from fastapi import Body, FastAPI, HTTPException
  File "/usr/local/lib/python3.8/site-packages/fastapi/__init__.py", line 7, in <module>
    from .applications import FastAPI as FastAPI
  File "/usr/local/lib/python3.8/site-packages/fastapi/applications.py", line 4, in <module>
    from fastapi import routing
  File "/usr/local/lib/python3.8/site-packages/fastapi/routing.py", line 22, in <module>
    from fastapi.datastructures import Default, DefaultPlaceholder
  File "/usr/local/lib/python3.8/site-packages/fastapi/datastructures.py", line 3, in <module>
    from starlette.datastructures import URL as URL  # noqa: F401
  File "/usr/local/lib/python3.8/site-packages/starlette/datastructures.py", line 8, in <module>
    from starlette.concurrency import run_in_threadpool
  File "/usr/local/lib/python3.8/site-packages/starlette/concurrency.py", line 10, in <module>
    from typing_extensions import ParamSpec
ImportError: cannot import name 'ParamSpec' from 'typing_extensions' (/usr/local/lib/python3.8/site-packages/typing_extensions.py)`

### Operating System

Linux

### Operating System Details

NAME="Ubuntu"
VERSION="21.04 (Hirsute Hippo)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 21.04"
VERSION_ID="21.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=hirsute
UBUNTU_CODENAME=hirsute

### FastAPI Version

Cannot check version name since I cannot import fastapi, ran pip install

### Python Version

3.8

### Additional Context

I am running the Fast API code from within a cog docker. I saw starlette was updated in the new fastapi release and was wondering if the issue is from the update, since last week this was working in another colleagues system when he ran directly.

**Comment by so2bin** [comment]
Same to me with fastapi-0.76.0, down back to fastapi-0.74.1 is ok.

**Comment by tonkolviktor** [comment]
same to me, fastapi-0.75.* is good.
fastapi-0.76.0 is good on py3.6 but fails on py3.7

**Comment by Kludex** [comment, marked as answer]
https://github.com/tiangolo/fastapi/issues/4868#issuecomment-1119678681
